controller/Results.php

```diff
@@ -553,24 +553,23 @@ protected function getResultStorage($delivery)
 
     /**
      * Regroup item variables by attempt
+     *
      * @param array $variables
      * @param array $filterTypes
      * @return array
+     * @throws common_Exception
      */
-    protected function formatItemVariables($variables, $filterTypes)
+    protected function formatItemVariables(array $variables, array $filterTypes): array
     {
         $displayedVariables = $this->getResultsService()->filterStructuredVariables($variables, $filterTypes);
         $responses = ResponseVariableFormatter::formatStructuredVariablesToItemState($variables);
         $excludedVariables = array_flip(['numAttempts', 'duration']);
 
-        foreach ($displayedVariables as &$item) {
-            if (!isset($item['uri'])) {
-                continue;
-            }
-            $itemUri = $item['uri'];
-            $state = isset($responses[$itemUri][$item['attempt']])
-                ? array_diff_key($responses[$itemUri][$item['attempt']], $excludedVariables)
+        foreach ($displayedVariables as $itemKey => &$item) {
```

Comment:
It's very sad to see how much logic we keep in our controllers. Especially because we never create a test for them. :(